addons/account/models/account_move.py

```diff
@@ -342,7 +342,8 @@ def _onchange_currency(self):
         else:
             self.line_ids._onchange_currency()
 
-        self._recompute_dynamic_lines(recompute_tax_base_amount=True)
+        if self.currency_id:
```

Comment:
Maybe it would make sense to add this condition at the very beginning of the method (and `return`). As you mention, `currency_id` is required, so at some point we must execute the onchange with a currency set. I wouldn't be surprised if the calls to the other methods would cause some issues.